src/platform/silabs/EndpointQueueFilter.cpp

```diff
@@ -0,0 +1,141 @@
+#include "EndpointQueueFilter.h"
+#include <string.h>
+#include <support/CodeUtils.h>
+#include <support/logging/CHIPLogging.h>
+
+namespace chip {
+namespace Inet {
+
+using FilterOutcome = EndpointQueueFilter::FilterOutcome;
+using namespace chip::Logging;
+
+namespace {
+
+bool IsValidMdnsHostName(const chip::CharSpan & hostName)
+{
+    for (size_t i = 0; i < hostName.size(); ++i)
+    {
+        char ch_data = *(hostName.data() + i);
+        if (!((ch_data >= '0' && ch_data <= '9') || (ch_data >= 'A' && ch_data <= 'F')))
+        {
+            return false;
+        }
+    }
+    return true;
+}
+
+bool PayloadContains(const chip::System::PacketBufferHandle & payload, const chip::ByteSpan & byteSpan)
+{
+    if (payload->HasChainedBuffer() || payload->TotalLength() < byteSpan.size())
+    {
+        return false;
+    }
+    for (size_t i = 0; i <= payload->TotalLength() - byteSpan.size(); ++i)
+    {
+        if (memcmp(payload->Start() + i, byteSpan.data(), byteSpan.size()) == 0)
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+bool PayloadContainsCaseInsensitive(const chip::System::PacketBufferHandle & payload, const ByteSpan & name)
```

Comment:
What does "case insensitive" even mean for _bytes_?   This must be getting used on some fairly narrow range of input values, so:

1. It should probably take a CharsSpan, not a ByteSpan.
2. It should document its assumptions (e.g. that the input is an ASCII string?).